cin_validator/cin_validator_class.py

```diff
@@ -10,18 +14,42 @@
 
 class CinValidationSession:
     def __init__(self, filename, ruleset, issue_id=None) -> None:
-        # TODO detect filetype xml/csv/zip. check if the directory is a folder.
-        fulltree = ET.parse(filename)
-        root = fulltree.getroot()
-        self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
-
         self.ruleset = ruleset
         self.issue_id = issue_id
+        self.filename = filename
 
+        self.file_type_checker()
         self.create_error_report_df()
         self.create_json_report()
         self.select_by_id()
 
+    def file_type_checker(self):
+        fn = self.filename.name
+        if os.path.isfile(fn):
+            extension = pathlib.Path(fn).suffix
+            if extension[:4] == ".xml":
+                fulltree = ET.parse(fn)
+                root = fulltree.getroot()
+                self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
+                print("got here!")
+            elif extension[:4] == ".csv":
+                print(".csv support coming soon!")
+            elif extension[:4] == ".zip":
+                zf = zipfile.ZipFile(fn)
+                dfs = {
+                    text_file.filename[:-4]: pd.read_csv(zf.open(text_file.filename))
+                    for text_file in zf.infolist()
+                    if text_file.filename.endswith(".csv")
+                }
+                print(dfs)
+                self.data_files_obj = DataContainerWrapper(dfs)
+            else:
+                print(f"Filetype: {extension} not currently supported")
+        elif os.path.isdir(fn):
```

Comment:
Getting a list of all the files in a directory such that you can check their types and read them (as you did above for the zip)
https://stackoverflow.com/questions/3207219/how-do-i-list-all-files-of-a-directory
https://www.geeksforgeeks.org/how-to-read-multiple-text-files-from-folder-in-python/

Reply:
We can leave out directory handling since we won't be handling CSVs in the MVP version.